### PR TITLE
User interface

### PR DESCRIPTION
**Pull request by widzthedvloper**
This branch contains the basic flow of the Tic tac toe game.
The logic part of the game is not implemented, yet.
This branch only showcases our the users will interact with the game.

### CODE REVIEW

**Review by teekaytech — CHANGES_REQUESTED** [review]
## Status: Changes Required :warning: :vertical_traffic_light: 

### Hello team :wave:

__You've done really well in this milestone.
However, some changes are required before it can be approved. Please check my comments below and fix them as appropriate.__

**[ REQUIRED CHANGES ]**

- [x] The game should display the board after a player move (without computing which spots on the board are taken) 
- [x] Game should inform a player if the selected move is invalid (without computing what an invalid move is)
- You've managed to inform the player if a move is a win or draw move, now you should take care of invalid move too, without computing what an invalid move is.


* * *
**Please make the required changes and submit another review request when you are done.**

* * *

**Taofeek Olalere
[Twitter](https://twitter.com/ola_lere) | [Github](https://github.com/teekaytech) | @taofeekolalere on Slack**
__Happy Coding :smiley:__

**Review by GzimAsani — CHANGES_REQUESTED** [review]
### CHANGES REQUIRED  :stop_sign: 
### HELLO @widzthedvloper You did a great job implementing the changes from the last review :+1:  very nice work :+1:  but you have missed one thing
- [x]  Game should inform a player if the selected move is invalid (without computing what an invalid move is)

[OPTIONAL]
- [x]  A loop is needed in this game (recommended is `while` but anything else will do)

### **Please make the required changes and submit another review request when you are ready.**

<hr>
### if you have any question kindly reach me on slack @GzimAsani

### HAPPY CODING :computer:

**Review by NtwaliHeritier — CHANGES_REQUESTED** [review]
**Changes required**
Hello @widzthedvloper . Great work. However, some changes are required for the project to be approved.
- Add a loop that always displays the board before each move until the game ends.
- The game ends only after two entry. Kindly fix it. The maximum number of moves should be 9 (draw), if no win occurred.
-  For each move, display the name of the player whose turn it is to play.

Kindly fix those issue and ask for another review.
Feel free to contact me on slack(username: NTWALI Heritier).
Happy coding, You've got this👍

**Review by NtwaliHeritier — CHANGES_REQUESTED** [review]
**Changes required**
Hello Team,
Great work, but changes are to be made.
- The features required in the previous review are not made.
- As a hint, you can place logics in functions. for example(a display board function, ...). Check out the unending while loop [while(true)] and look for how to break out of it. This can help you to know when to get out of a loop. i.e: In case of a win or a draw.

Make the requested changes and ask for another review.

**Comment by widzthedvloper** [comment]
@NtwaliHeritier  this branch is only for the user interface, I don't need to put any logic yet into the project. what I'm trying to do now is to show how the game will interact with the user, also I mention those things in the comments I put in the code. I also simulate a way to break out the while loop. the logic goes for milestone 3 and I'm currently doing milestone two I don't understand why should I call any method yet as per the requirement of that milestone I only need to show the basic flow of how the game will interact with each player.

File: bin/main.rb
Comment (MarilenaRoque) [inline bin/main.rb]: If you use the code to puts lines directly, it can not be modified to change when the user provides input. If you store these lines on an array, for example, you can easily add the logic to replace the '-' for a symbol 'X' or 'O' to indicate the move.

File: bin/main.rb
Comment (MarilenaRoque) [inline bin/main.rb]: - The 'while' should check a variable that defines a winner or has a maximum number of moves for a draw, for example. You can fake the behavior of the variable while you don't have the logic, but it needs to be ready to receive the logic.

**Review by MarilenaRoque — CHANGES_REQUESTED** [review]
# Hello, @widzthedvloper! 🚀 

### I recognize that you already made a great effort to complete that milestone, and guidelines can be quite confusing sometimes. 👍🏻 
Although we have strict requirements that need to be completed before approval.

☑️ This milestone doesn't need logic for the game functionality. For example, to check the winner or to replace the board. 
### ✔️ **But you do need to build the interface in such a way that the only thing missing will be the logic behind.**. I mean:  ----> THE INTERFACE SHOULD BE READY TO RECEIVE THE LOGIC <---. 

Please check my comments below, to complement my teammates' explanations with examples. 

It can be frustrating when you receive multiple requirements for changes. I know. So I recommend that when it happens you contact one of the TSE members in Slack directly. We will be happy to provide better details about the requirements. 😃

File: bin/main.rb
Comment (MarilenaRoque) [inline bin/main.rb:44]: - Don't forget that when a spot already has an 'X' or 'O' it should be an invalid move too.

**Review by MarilenaRoque — APPROVED** [review]
# Congratulations on complete the changes required!!! 🥇 

This pull request is approved and you are good to go for the next step of this project.

I am kindly adding a few reminders below to help with the implementation of the logic in the next milestone.

### Happy Coding!!! 💻 

**_Mari Roque_**

**Pull request merged**
